modules/core/src/main/java/org/apache/ignite/internal/util/PendingComparableValuesTracker.java

```diff
@@ -61,7 +62,7 @@ public class PendingComparableValuesTracker<T extends Comparable<T>, R> implemen
     private volatile boolean closeGuard;
 
     /** Busy lock to close synchronously. */
-    private final IgniteSpinBusyLock busyLock = new IgniteSpinBusyLock();
+    private final IgniteStripedReadWriteLock busyLock = new IgniteStripedReadWriteLock();
```

Comment:
It violates a contract of a busyLock, it should not be possible to "enterBusy" previously "blocked" lock, however it's possible to busyLock.readLock.lock() after busyLock.wrireLock().lock(). 
PendingComparableValuesTracker class is an utility one, it's not valid to expect that close and e.g. update will be called within different treads even if it's true for now.

Reply:
I've created a ticket to fix this
https://issues.apache.org/jira/browse/IGNITE-23716